snackpub/src/index.ts

```diff
@@ -12,13 +15,17 @@ import type {
 
 const debug = require('debug')('snackpub');
 
-function registerShutdownHandlers(server: Server) {
+function registerShutdownHandlers(
+  server: Server,
+  redisClient: ReturnType<typeof createClient> | null
+) {
   const shutdown = async (signal: ShutdownSignal) => {
     console.log(
       `Received ${signal}; the HTTP server is shutting down and draining existing connections`
     );
     await closeServerAsync(server);
-    await maybeCloseRedisConnectionsAsync();
+    await closeRedisAdapterAsync();
+    await redisClient?.quit();
```

Comment:
I think this shutdown routine needs to be more graceful. Specifically, we need to reject all new connections so that we drain connections from the server. This is where a readiness probe or health check is especially helpful. A failing health check tells the load balancer to stop sending new traffic to an unhealthy server instance. Additionally, we should notify the clients that we're going to terminate the connections and that they should try to reconnect to another instance. This can happen in another PR but I think it is important for service reliability.

That said, I think the code here does a good job of making sure to stop the WebSocket server before terminating the connections to Redis. We could also use Promise.all to terminate both Redis connections simultaneously.

Reply:
will follow up readiness in a separated pr.

> We could also use Promise.all to terminate both Redis connections simultaneously.
kinda afraid to do it simultaneously, so to terminate the subscriber client first and then the main redis client.